src/ib_py/cogs/channelorder.py

```diff
@@ -1,31 +1,209 @@
+
+    def draw_embed(self, title: str, fields: list[tuple[str, str, bool]], color=discord.Color.blue()):
         """
-        raise NotImplementedError("Command requires implementation and permission set-up.")
+        Helper function to draw an embed with the given title, fields, and color.
+        
+        Args:
+            title (str): The title of the embed.
+            fields (list[tuple[str, str, bool]]): A list of fields where each field is a tuple (name, value, inline).
+            color (discord.Color, optional): The color of the embed. Defaults to discord.Color.blue().
 
-    @channelorder.command()
-    async def snapshot(self, ctx: commands.Context):
+        Returns:
+            discord.Embed: The constructed embed object.
         """
-        Save the arrangement of channels in a category.
+        embed = discord.Embed(title=title, color=color)
+        for name, value, inline in fields:
+            embed.add_field(name=name, value=value, inline=inline)
+        return embed
+
+    def create_snapshot_embed(self, category, channels):
         """
-        raise NotImplementedError("Command requires implementation and permission set-up.")
+        Helper function to create a snapshot embed for a given category.
+        Channels are listed on separate lines and mentioned using <#id>.
+        """
+        fields = []
+        if channels:
+            # Mention each channel on a new line
+            channel_mentions = "\n".join(f"<#{ch.id}>" for ch in channels)
+        else:
+            channel_mentions = "None"
+
+        fields.append(("Channels", channel_mentions, False))
+        return self.draw_embed(title=f"📸 Snapshot for {category.name}", fields=fields)
+
+
+    @channelorder.command(name="snapshot")
+    @commands.has_permissions(administrator=True)
+    async def snapshot(self, ctx: commands.Context, *, category_input: str):
+        """Take a snapshot of all channels (text, voice, forum) in the given category."""
+        guild = ctx.guild
+        category = self.get_category(guild, category_input)
+
+        if not category:
+            await ctx.reply("Invalid category ID or name.", ephemeral=True)
+            return
+
+                # Get all types of channels
+        voice_channels = category.voice_channels
+        # Sort all channels by position
+        all_channels_sorted = sorted(category.channels, key=lambda c: (c.position, c.id))
+
+        # Separate and recombine: text+forum first, then voice (voices keep relative order)
+        text_forum_channels = [ch for ch in all_channels_sorted if ch.type in (discord.ChannelType.text, discord.ChannelType.forum)]
+        voice_channels = [ch for ch in all_channels_sorted if ch.type == discord.ChannelType.voice]
+
+        # Combine: text+forum first, then voice (voice order preserved among themselves)
+        all_channels = text_forum_channels + voice_channels
+
+        embed = self.create_snapshot_embed(category, all_channels)
```

Comment:
According to the [docs](https://discordpy-reborn.readthedocs.io/en/latest/api.html?highlight=mention#discord.CategoryChannel.channels), `category.channel` should already contain the channels in order (with voice channels after text channels). Afaik, there's no need to separate out channel types, you could do
```py
embed = self.create_snapshot_embed(category, category.channels)
```
This would have the added benefit of managing all channel types by default (e.g. stage channels and other types would not need to be separately added in).